File: aprendizado/exercicios/AndroidApp/app/src/main/java/com/example/androidapp/GradeCalculatorActivity.java
package com.example.androidapp;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class GradeCalculatorActivity extends AppCompatActivity {

    private EditText grade1Input;
    private EditText grade2Input;
    private EditText grade3Input;
    private Button calculateButton;
    private TextView resultTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_grade_calculator);

        grade1Input = findViewById(R.id.grade1Input);
        grade2Input = findViewById(R.id.grade2Input);
        grade3Input = findViewById(R.id.grade3Input);
        calculateButton = findViewById(R.id.calculateButton);
        resultTextView = findViewById(R.id.resultTextView);

        calculateButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calculateAverage();
            }
        });
    }

    private void calculateAverage() {
        String grade1Str = grade1Input.getText().toString();
        String grade2Str = grade2Input.getText().toString();
        String grade3Str = grade3Input.getText().toString();

        if (!grade1Str.isEmpty() && !grade2Str.isEmpty() && !grade3Str.isEmpty()) {
            double grade1 = Double.parseDouble(grade1Str);
            double grade2 = Double.parseDouble(grade2Str);
            double grade3 = Double.parseDouble(grade3Str);

            double average = (grade1 + grade2 + grade3) / 3;
            String result;

            if (average >= 6) {
                result = "Aprovado! Média: " + average;
            } else {
                result = "Reprovado! Média: " + average;
            }

            resultTextView.setText(result);
        } else {
            resultTextView.setText("Por favor, preencha todas as notas.");
        }
    }
}

File: aprendizado/exercicios/MyAndroidApp/app/src/main/java/com/example/myandroidapp/GradeCalculatorActivity.kt
package com.example.myandroidapp

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class GradeCalculatorActivity : AppCompatActivity() {

    private lateinit var grade1EditText: EditText
    private lateinit var grade2EditText: EditText
    private lateinit var grade3EditText: EditText
    private lateinit var calculateButton: Button
    private lateinit var resultTextView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_grade_calculator)

        grade1EditText = findViewById(R.id.grade1EditText)
        grade2EditText = findViewById(R.id.grade2EditText)
        grade3EditText = findViewById(R.id.grade3EditText)
        calculateButton = findViewById(R.id.calculateButton)
        resultTextView = findViewById(R.id.resultTextView)

        calculateButton.setOnClickListener {
            calculateAverage()
        }
    }

    private fun calculateAverage() {
        val grade1 = grade1EditText.text.toString().toFloatOrNull()
        val grade2 = grade2EditText.text.toString().toFloatOrNull()
        val grade3 = grade3EditText.text.toString().toFloatOrNull()

        if (grade1 != null && grade2 != null && grade3 != null) {
            val average = (grade1 + grade2 + grade3) / 3
            val result = if (average >= 6) "Approved" else "Failed"
            resultTextView.text = "Average: $average\nResult: $result"
        } else {
            resultTextView.text = "Please enter valid grades."
        }
    }
}